src/framegrab/grabber.py

```diff
@@ -99,16 +99,41 @@ def create_grabbers(configs: List[dict]) -> dict:
                 f"Provided camera names: {names}"
             )
 
-        grabbers = {}
+        # Create the grabbers
+        grabber_list = []
         for config in configs:
-            grabber = FrameGrabber.create_grabber(config)
-            name = grabber.config["name"]
-            grabbers[name] = grabber
+            grabber = FrameGrabber.create_grabber(config, autogenerate_name=False)
+            grabber_list.append(grabber)
+
+        grabbers = FrameGrabber.grabbers_to_dict(grabber_list)
 
         return grabbers
 
     @staticmethod
-    def create_grabber(config: dict):
+    def grabbers_to_dict(grabber_list: list) -> dict:
+        """Converts a list of FrameGrabber objects into a dictionary where the keys are the camera names.
+        Sorts the grabbers by serial_number to make sure they always come up in the same order.
+        Autogenerates names for any unnamed grabbers.
+        """
+
+        # Sort the grabbers by serial_number to make sure they always come up in the same order
+        grabber_list = sorted(grabber_list, key=lambda grabber: grabber.config.get("id", {}).get("serial_number", ""))
```

Comment:
Since we are sorting them here, can we remove the sorting we do in `create_grabbers` method? For now we are sorting one extra time, which I don't think is necessary.